client/src/components/ReviewTile.js

```diff
@@ -1,13 +1,50 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 
-const ReviewTile = ({ review }) => {
-  const { subject, body, rating } = review
+const ReviewTile = ({ review, castVote, currentUser }) => {
+  const { id, author, subject, body, rating, votes } = review
+  const { score, currentUserVote } = votes
+  const [voteState, setVoteState] = useState(null)
+
+  const handleButtonClick = async (event) => {
+    if (currentUser) {
+      const { currentTarget } = event
+      const value = parseInt(currentTarget.value)
+
+      if (voteState & value === voteState) {
+        setVoteState(null)
+      } else {
+        setVoteState(value)
+      }
+      await castVote(id, value)
+    } else {
+      alert("You must be signed in to vote")
+    }
+  }
+
+  useEffect(() => {
+    setVoteState(currentUserVote.voteValue)
+  }, [review])
```

Comment:
we shouldnt need useEffect for this. instead, just set the default value of voteState to currentUserVote.voteValue